### PR TITLE
Team 6 game area refactor

### PR DESCRIPTION
**Pull request by CameronBadman**
# Description

made the MainGameArea componant no longer inherit GameArea instead it is called GameController, and uses dependancy injection to work with GameArea 

Fixes / Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Game runs 
- [x] Boss Room works
- [x] levels system works 
- [x] all tests pass 
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by IanP96 — DISMISSED** [review]
Mostly looks good but would be worth merging in main again and getting review from the character team, also surely add the boss music. There's functions in main that will do this

**Comment by vsinh14** [comment]
What benefit does this PR add to the studio as a whole to change a lot of core things 3 days before submission (does it resolve bugs etc to warrant this extra overhead for the whole studio)

**Comment by CameronBadman** [comment]
@vsinh14 it actually doesn't change much as a studio, the GameArea singleton will work as intended for 80 to 90% of people

File: source/core/src/main/com/csse3200/game/areas/BossRoom.java [hunk 1]
```diff
@@ -59,21 +59,15 @@ public BossRoom(NPCFactory npcFactory,
...
         this.area = area;
         this.player = player;
         spawnStairs(player, area);
 
     }
 
-    /**
-     *
-     * @param player  the player entity of the game
-     * @param area   the main game area for stairs
```
Comment (vsinh14) [inline source/core/src/main/com/csse3200/game/areas/BossRoom.java:73]: bring back comments

**Review by vsinh14 — COMMENTED** [review]
Please make the requested changes

**Comment by CameronBadman** [comment]
@IanP96 boss music and normal music added

**Review by vsinh14 — APPROVED** [review]
Tested and all requested changes made

File: source/core/src/main/com/csse3200/game/areas/GameArea.java [hunk 2]
```diff
@@ -102,4 +116,122 @@ public CopyOnWriteArrayList<Entity> getListOfEntities() {
...
+
+        Music music = resourceService.getAsset(BACKGROUND_MUSIC, Music.class);
+        music.setLooping(true);
+        if (!UserSettings.get().mute) {
+            music.setVolume(UserSettings.get().musicVolume);
+        } else {
+            music.setVolume(0);
+        }
+        music.play();
+    }
```
[the remark this answers is not in the document]
Comment (vsinh14) [reply source/core/src/main/com/csse3200/game/areas/GameArea.java]: has been added back

**Review by SIN-Willett — CHANGES_REQUESTED** [review]
unsafe op

**Review by SIN-Willett — APPROVED** [review]
better

**Pull request merged**